### PR TITLE
Create /chat page with existing social features

### PR DESCRIPTION
**Pull request by mrhegemon**
## Summary

**FEATURES**

This PR includes the following:

- New /chat route using tailwind
- New FeathersHooks utility for Reactive feathers requests and caching
- replaces parties and groups with channels
- media instance connection for world and channels unified

**DATABASE MIGRATIONS**

- `party`, `party_user`, `group`, `group_user`, `group_user_rank` `channel_type` tables have been dropped
- `group:read` `group:write` `party:read` `party:write` scopes have been replaced with `channel:read` and `channel:write`
- `channel_user` table has been added
- `group` and `party` invite types are now `channel`
- `group` and `party` invite email subject fields collapsed into `channel`

## References

closes #_insert number here_

## Checklist
- [ ] If this PR is still a WIP, convert to a draft
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] [ensure all checks pass](https://github.com/etherealengine/etherealengine/wiki/Testing-&-Contributing)
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewer

## QA Steps

- ensure admin panel channels page works correctly, populating data and search query 
- ensure socials work, messaging friends and starting and ending media sessions with them

### CODE REVIEW

File: packages/client-core/package.json [hunk 1]
```diff
@@ -78,7 +78,14 @@
     "tick-tock": "^1.0.0",
     "typescript": "5.0.2",
     "util": "^0.12.5",
-    "uuid": "9.0.0"
+    "uuid": "9.0.0",
+    "pino": "^8.11.0",
```
Comment (barankyle) [inline packages/client-core/package.json:82]: Most of these new packages don't appear to be used anywhere in this PR. Please review whether they are needed and remove them if they are not.

File: packages/client-core/src/admin/components/Setting/Email.tsx [hunk 2]
```diff
@@ -211,14 +211,7 @@ const Email = () => {
             name="group"
```
Comment (barankyle) [inline packages/client-core/src/admin/components/Setting/Email.tsx:211]: Does this need to have the other usages of 'group' renamed? And change the naming in admin.json?

**Comment by speigg** [comment]
/chat route 
- page is not responsive, seems to have a fixed width
- UI seems generally broken/buggy (pretty much nothing works? am I meant to be reviewing this route?)

/location route
- Sometimes, sending a message does not show the message to the recipient immediately (have to exit the message window and return to see it, or send back a message to the other user, and then the previously received messages show up)
- why does clicking on the chat icon next to the user in the "Friends" menu open up a group chat? Is there only one group chat shared between all your friends?
- There are no user names in the group chat, no way to know who is saying what
- layout / style of the chat window looks unpolished

**Comment by HexaField** [comment]
> /location route
> 
> * Sometimes, sending a message does not show the message to the recipient immediately (have to exit the message window and return to see it, or send back a message to the other user, and then the previously received messages show up)
> * why does clicking on the chat icon next to the user in the "Friends" menu open up a group chat? Is there only one group chat shared between all your friends?
> * There are no user names in the group chat, no way to know who is saying what
> * layout / style of the chat window looks unpolished

All these location bugs have been fixed. The styling still could use some work, but we need this PR in for the added functionality ASAP.

**Review by barankyle — APPROVED** [review]

**Pull request merged**